src/hdmf_zarr/utils.py

```diff
@@ -28,12 +48,18 @@ def __init__(self):
         super().__init__()
 
     @classmethod
-    def __write_chunk__(cls, dset, data):
+    def __write_chunk__(cls, dset: HDMFDataset, data: DataChunkIterator):
         """
         Internal helper function used to read a chunk from the given DataChunkIterator
         and write it to the given Dataset
         :param dset: The Dataset to write to
+        :type dset: HDMFDataset
         :param data: The DataChunkIterator to read from
+        :type data: DataChunkIterator
+        :param buffer_index: The index of the next chunk within the DataChunkIterator, optional.
+        Only used with multiprocessing for serializing the selections of the iteration.
+        Default is to perform direct iteration on the in-memory DataChunkIterator for the single process.
+        :type buffer_index: integer or None
```

Comment:
buffer_index is not listed in the method kwargs above

Reply:
Ah, that was leftover from early attempts. Passing the buffer selection explicitly turned out to be more efficient

Reply:
We also decided not to alter `__write_chunk__` at all in this case

Reply:
Got it. I'll remove that from the docstring